test/specs/file_test.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:code_builder/code_builder.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('File', () {
+    const $linkedHashMap = const Reference('LinkedHashMap', 'dart:collection');
+    test('should emit a source file', () {
+      expect(
+        new File((b) => b
+          ..directives.add(new Directive.import('dart:collection'))
```

Comment:
Is the assumption that you always have to add the imports manually? Can you add a test which uses allocator.imports instead?

Also, maybe a test case for prefixing imports?